### ISSUE TITLE
[Enhancement] A11y improvements for Calendar

### ISSUE THREAD

**Issue by RasmusKjeldgaard** [kirbydesign/designsystem#3681]
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the enhancement
Based on the [date picker dialog pattern](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/datepicker-dialog/), but without the input field functionality, the following improvements should be considered to improve the accessibility of the calendar: 

- [x] Keyboard support for moving around the grid of dates - as detailed under **_Date Picker Dialog: Date Grid_** in the above link.
- [x] (Probably optional) Shortcut keys for changing the month and year displayed in the calendar.
- [x] The calendar heading displaying the month and year should be marked as a live region so screen reader users get feedback from the buttons and keyboard commands that change the month and year.
- [ ] Possibly keyboard help to convey arrow-key functionality when moving aroind the grid. A live region can be used to announce it to screen reader users when focus moves into the calendar grid.
- [x] Day names and column headers are abbreviated to single characters in the design. However, this makes it more difficult for screen reader users to understand the day names. So, full day names should be provided to assistive technologies in the HTML abbr attribute on the column headers, enabling screen readers to announce the full names when users navigate the grid. 

## Have you considered any alternatives?
- Some calendars, like [ion-datetime](https://ionicframework.com/docs/api/datetime#basic-usage) use a grid of buttons. 
- [Cally](https://wicky.nillia.ms/cally/) use a table, but with the addition of buttons for each selectable day. This approach is outlined here: https://wicky.nillia.ms/cally/accessibility/#approach

## Are there any additional context?

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#the-process-of-contributing) correctly.

### Refinement

- [ ] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#ux-refinement)
- [ ] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](../CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".
